faiss/impl/pq4_fast_scan.cpp

```diff
@@ -122,30 +122,59 @@ void pq4_pack_codes_range(
     }
 }
 
+uint8_t get_address(size_t& bbs, size_t& vector_id, size_t& sq, bool& shift) {
```

Comment:
passing arguments with & does not make sense here

Reply:
for "shift", it is mandatory because we need shift in the output.
for other args, I made that to avoid creating new 3 variables with 3 new memory locations  but after rethink, it is not much improvement